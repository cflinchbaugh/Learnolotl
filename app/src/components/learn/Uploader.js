import React, { Component } from 'react';

class Uploader extends Component {
    handleFileUpload(e) {
        let files = e.target.files,
            i = 0;

        for ( ; i < files.length; i++) { //for multiple files          
            ( (file) => {
                let reader = new FileReader();

                reader.onload = (e) => {
                    // Async code, as files complete loading we handle them individually,
                        // do not try to aggregate, just handle the new file
                        // by parsing it, grabbing the results, and sending the data up

                    let parsedData = this._validateJSON(e.target.result),
                        processedUploadData;
                        
                    if (parsedData) {
                        processedUploadData = {
                            fileId: parsedData.id,
                            fileDataArr: parsedData.results,
                            format: parsedData.format
                        };
                        
                        this.props.handleFileUpload(processedUploadData);
                    }   
                }
                             
                reader.readAsText(file, "UTF-8");
            })(files[i]);
        }
    }

    _validateJSON(data) {
        try {
            let parsedData = JSON.parse(data);
            
            return parsedData;
        } catch(e) {
            
            alert('Invalid file upload!  File must be valid .json.  File ignored.');
            
            return null;
        }
    }

    render() {
        return (
            <div>
                <h1>Uploader</h1>

                <input 
                    type="file" 
                    name="file" 
                    onChange={(e)=>this.handleFileUpload(e)} 
                    multiple="multiple"/>
            </div>
        );
    }
}

export default Uploader;