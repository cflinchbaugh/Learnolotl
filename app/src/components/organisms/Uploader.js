import React, { Component } from 'react';

class Uploader extends Component {
    constructor(props) {
        super(props);

        this.state = {
            defaultData: true
        }
    }

    handleFileUpload(e) {
        let files = e.target.files,
            i = 0;

        for ( ; i < files.length; i++) { //for multiple files          
            ( (file) => {
                let reader = new FileReader();

                reader.onload = (e) => {
                    // Async code, as files complete loading we handle them individually,
                        // do not try to aggregate, just handle the new file
                        // by parsing it, grabbing the results, and sending the data up

                    let processedFileData = [],
                        j,
                        fileDataArr;

                    fileDataArr = JSON.parse(e.target.result).results;

                    for (j = 0; j < fileDataArr.length; j++) {
                        processedFileData.push(fileDataArr[j]);
                    }

                    this.props.handleFileUpload(processedFileData);
                }
                             
                reader.readAsText(file, "UTF-8");
            })(files[i]);
        }
    }

    render() {
        return (
            <div>
                <h1>Uploader</h1>

                <input 
                    type="file" 
                    name="file" 
                    onChange={(e)=>this.handleFileUpload(e)} 
                    multiple="multiple"/>
            </div>
        );
    }
}

export default Uploader;