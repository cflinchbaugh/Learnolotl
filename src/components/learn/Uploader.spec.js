import React from 'react';
import {shallow} from 'enzyme';
import Uploader from './Uploader';
import ShallowRenderer from 'react-test-renderer/shallow';

describe('Uploader', () => {
    it('renders and matches snapshot', () => {
        const renderer = new ShallowRenderer();

        const uploaderData = {
            
        }

        const tree = renderer.render(
            <Uploader {...uploaderData}/>
        );

        expect(tree).toMatchSnapshot();
    });
});