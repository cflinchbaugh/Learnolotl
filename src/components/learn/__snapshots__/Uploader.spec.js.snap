// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Uploader renders and matches snapshot 1`] = `
<div>
  <input
    label="Uploader"
    multiple="multiple"
    name="file"
    onChange={[Function]}
    type="file"
  />
</div>
`;
